src/NServiceBus.RavenDB/Outbox/OutboxPersister.cs

```diff
@@ -109,16 +108,16 @@ public async Task SetAsDispatched(string messageId, ContextBag options, Cancella
                 if (useClusterWideTransactions)
                 {
                     var compareExchangeKey = $"rvn-atomic/{outboxRecordId}";
-                    var outboxRecord = await session.LoadAsync<OutboxRecord>(outboxRecordId, includes => includes.IncludeCompareExchangeValue(outboxRecordId), cancellationToken)
-                        .ConfigureAwait(false);
-                    var compareExchangeValue = await session.Advanced.ClusterTransaction.GetCompareExchangeValueAsync<CompareExchangeValue<string>>(compareExchangeKey, cancellationToken)
-                        .ConfigureAwait(false);
+                    var outboxRecordLazy = session.Advanced.Lazily.LoadAsync<OutboxRecord>(outboxRecordId, cancellationToken);
+                    var compareExchangeValueLazy = session.Advanced.ClusterTransaction.Lazily.GetCompareExchangeValueAsync<string>(compareExchangeKey, cancellationToken);
```

Comment:
```suggestion
                    var compareExchangeValueLazy = session.Advanced.ClusterTransaction.Lazily.GetCompareExchangeValueAsync<string>(outboxLockCompareExchangeKey , cancellationToken);
```

Reply:
Sorry the PR was set to auto merge on approval :D https://github.com/Particular/NServiceBus.RavenDB/pull/1017